.github/workflows/deploy.yml

```
@@ -65,7 +71,10 @@
   # Alternative: Deploy backend to Render
   deploy-backend-render:
     runs-on: ubuntu-latest
-    if: github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.head_branch == 'main' && secrets.DEPLOY_TARGET != 'railway'
+    if: ${{ github.ref == 'refs/heads/main' && secrets.DEPLOY_TARGET != 'railway' }}
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)